### PR TITLE
Ndon/org

### PR DESCRIPTION
**Pull request by vysahjk**

### CODE REVIEW

File: Babylon/commands/api/organizations/security/update.py [hunk 1]
```diff
@@ -22,33 +18,24 @@
...
+    type=str,
     required=True,
+    default="viewer",
+    help="Role RBAC",
 )
-@output_to_file
-@inject_context_with_resource({'api': ['organization_id', 'url']})
-def update(context: Any, azure_token: str, security_file: pathlib.Path) -> CommandResponse:
+@option("--email", "email", type=str, required=True, help="Email valid")
+@argument("id", type=str)
```
Comment (esasova) [inline Babylon/commands/api/organizations/security/update.py:31]: I can't see the difference between email and id, isn't the same thing ?

File: Babylon/commands/api/organizations/security/__init__.py [hunk 2]
```diff
@@ -13,7 +12,6 @@ def security():
 list_commands = [
     add,
     get,
-    update,
 ]
```
[the remark this answers is not in the document]
Comment (vysahjk) [reply Babylon/commands/api/organizations/security/__init__.py:15]: oups!

File: Babylon/commands/api/organizations/security/add.py [hunk 3]
```diff
@@ -24,23 +24,22 @@
...
-                             azure_token,
-                             type="POST",
-                             data=details)
-    if response is None:
-        return CommandResponse.fail()
-    logger.info(SUCCESS__RBAC_UPDATED(org_id))
-    return CommandResponse.success(details, verbose=True)
+    service_state = state["services"]
+    service = ApiOrganizationSecurityService(azure_token=azure_token, state=service_state)
+    details = json.dumps(obj={"id": email, "role": role}, indent=2, ensure_ascii=True)
```
[the remark this answers is not in the document]
Comment (vysahjk) [reply Babylon/commands/api/organizations/security/add.py:43]: oups!

File: Babylon/services/organizations_service.py [hunk 4]
```diff
@@ -15,70 +13,87 @@
 
 class OrganizationsService:
```
Comment (esasova) [inline Babylon/services/organizations_service.py:14]: all other services use singular : Datase**t**Service, Solutio**n**Service etc :slightly_smiling_face:

File: Babylon/commands/api/organizations/security/add.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (vysahjk) [reply Babylon/commands/api/organizations/security/add.py:43]: @option("--email", "email", type=str, required=True, help="Email valid")

File: Babylon/commands/api/organizations/security/update.py [hunk 5]
```diff
@@ -22,33 +18,24 @@
...
-                          azure_token,
-                          type="POST",
-                          data=detail)
-        except Exception as exp:
-            logger.error(exp)
-            return CommandResponse.fail()
-    logger.info(f"Successfully updated organization {organization_id} RBAC")
-    return CommandResponse.success()
+    service_state = state["services"]
+    details = json.dumps({"id": email, "role": role})
```
[the remark this answers is not in the document]
Comment (vysahjk) [reply Babylon/commands/api/organizations/security/update.py:38]: both are mandatory

File: Babylon/commands/api/organizations/security/add.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (vysahjk) [reply Babylon/commands/api/organizations/security/add.py:43]: both are mandatory

File: Babylon/commands/api/organizations/security/add.py [hunk 6]
```diff
@@ -24,23 +24,22 @@
...
-@option("--role", "role", type=QueryType(), required=True, default="viewer", help="Role RBAC")
-@option("--email", "email", type=QueryType(), required=True, help="Email valid")
-@inject_context_with_resource({"api": ['url', 'organization_id']})
-def add(context: Any, azure_token: str, role: str, email: str, org_id: str = None) -> CommandResponse:
+@option(
+    "--role",
+    "role",
+    type=str,
+    required=True,
+    default="viewer",
```
[the remark this answers is not in the document]
Comment (vysahjk) [reply Babylon/commands/api/organizations/security/add.py:32]: i will remove the default value 👍🏽 thanks!

File: Babylon/commands/api/organizations/security/service/api.py [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+
+from logging import getLogger
+from Babylon.utils.environment import Environment
+from Babylon.utils.request import oauth_request
+
+logger = getLogger("Babylon")
+env = Environment()
+
+
+class ApiOrganizationSecurityService:
```
[the remark this answers is not in the document]
Comment (vysahjk) [reply Babylon/commands/api/organizations/security/service/api.py:11]: ??

File: Babylon/commands/api/organizations/delete.py [hunk 8]
```diff
@@ -19,18 +19,24 @@
...
-    organizations_service = OrganizationsService(state['services'], azure_token)
-    response = organizations_service.delete(id, force_validation)
+    service_state = state["services"]
+    service_state["api"]["organization_id"] = (organization_id or state["services"]["api"]["organization_id"])
+    if not force_validation and not confirm_deletion("organization", service_state["api"]["organization_id"]):
+        return CommandResponse.fail()
+    organizations_service = OrganizationsService(state=service_state, azure_token=azure_token)
+    response = organizations_service.delete()
     if response is None:
         return CommandResponse.fail()
```
[the remark this answers is not in the document]
Comment (vysahjk) [reply Babylon/commands/api/organizations/delete.py:33]: ??

File: Babylon/commands/api/organizations/security/update.py (hunk 1 above)
Comment (vysahjk) [reply Babylon/commands/api/organizations/security/update.py:31]: id is an email too, we can rename it to `identity_id`

File: Babylon/commands/api/organizations/delete.py [hunk 9]
```diff
@@ -19,18 +18,22 @@
...
-    try:
-        response.json()['organization_id']
-    except KeyError:
-        logger.error("Organization id is missing in response")
-
-    logger.info(SUCCESS_DELETED("organization", response.json()['organization_id']))
+    if response:
+        org_id = service_state["api"]["organization_id"]
+        logger.info(f'Organization {org_id["organization_id"]} successfully deleted')
```
Comment (esasova) [inline Babylon/commands/api/organizations/delete.py:33]: still no need of `org_id["organization_id"]` :slightly_smiling_face:

**Review by Leopold-Cramer — APPROVED** [review]

**Review by esasova — APPROVED** [review]
just one thing, LGTM :+1:

**Pull request merged**